lib/go/thrift/simple_server.go

```diff
@@ -192,13 +195,27 @@ func (p *TSimpleServer) innerAccept() (int32, error) {
 		return 0, err
 	}
 	if client != nil {
-		p.wg.Add(1)
+		ctx, cancel := context.WithCancel(context.Background())
+		p.wg.Add(2)
```

Comment:
style nit: it's probably less error prone to use `wg.Add(1)` right before every `go`, e.g. write it this way:

```go
ctx, cancel := context.WithCancel(context.Background())

p.wg.Add(1)
go func() {
  defer p.wg.Done()
  // actual work
}()

p.wg.Add(1)
go func() {
  defer p.wg.Done()
  // actual work
}()
```

this avoids the potential bug that someone adds another goroutine in the future but forgot to change the `2` to `3`.

also style-wise, it's really better to always immediately do `defer cancel()` right after `ctx, cancel := ...` (which is a style guide to avoid resource leaks), which is not possible in the current 2 separated goroutines way but possible if you wrap the second goroutine as a goroutine inside the first one (which was what I suggested in https://github.com/apache/thrift/pull/2497#discussion_r779887382 :) )

I don't feel too strongly for either of the nits here, though.

Reply:
ctx cancel is done by close a chan
```golang
func (c *cancelCtx) cancel(removeFromParent bool, err error) {
	if err == nil {
		panic("context: internal error: missing cancel error")
	}
	c.mu.Lock()
	if c.err != nil {
		c.mu.Unlock()
		return // already canceled
	}
	c.err = err
	d, _ := c.done.Load().(chan struct{})
	if d == nil {
		c.done.Store(closedchan)
	} else {
		close(d)
	}
	for child := range c.children {
		// NOTE: acquiring the child's lock while holding parent's lock.
		child.cancel(false, err)
	}
	c.children = nil
	c.mu.Unlock()

	if removeFromParent {
		removeChild(c.Context, c)
	}
}
```
And yes, if the ctx is caneled, it will not close the inner channel again, and will do nothing and return. So we still have to create a new ctx after stop the server, because the ctx.Done() channel is already stopped.

divide p.wg.Add(2) into two p.wg.Add(1), it seems ok for now